src/hashtree.erl

```diff
@@ -958,21 +1058,19 @@ iterate({ok, K, V}, IS=#itr_state{itr=Itr,
                                final_acc=[{Segment, F(Acc)} | FinalAcc],
                                prefetch=true},
             iterate(iterator_move(Itr, prefetch), IS2);
-        {Id, NextSeg, [NextSeg|Remaining], _} ->
-            %% A previous prefetch_stop left us at the start of the
-            %% next interesting segment.
+        {Id, _, [NextSeg | Remaining], true} ->
+            %% Pointing at uninteresting segment, but need to halt the
+            %% prefetch to ensure the iterator can be reused
+
+            %% Do not update segment/final_acc
```

Comment:
I think this comment should probably be removed, since now we do in fact update `segment` and `final_acc` here.